assert.go

```diff
@@ -794,3 +795,19 @@ func AssertDecreasing(t testRunner, object interface{}, msg ...interface{}) {
 
 	internal.AssertCompareHelper(t, object, -1, msg...)
 }
+
+func AssertRegexp(t testRunner, regex string, txt string, msg ...interface{}) {
+	if test, ok := t.(helper); ok {
+		test.Helper()
+	}
+
+	internal.AssertRegexpHelper(t, regex, txt, true, msg...)
+}
+
+func AssertNotRegexp(t testRunner, regex string, txt string, msg ...interface{}) {
```

Comment:
We should probably accept an interface as regex, since the user could use a string or an already existing regexp. The text might also be an interface, that is later converted to text using `fmt.Sprint()`, so that the user can use any type of input.